src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClientManager.cs

```diff
@@ -103,5 +104,17 @@ public bool UpdateSyncToken(Uri endpoint, string syncToken)
 
             return false;
         }
+
+        public Uri GetEndpointForClient(ConfigurationClient client)
+        {
+            if (client == null)
+            {
+                throw new ArgumentNullException(nameof(client));
+            }
+
+            ConfigurationClientWrapper currentClient = _clients.FirstOrDefault(c => c.Client == client);
+            
+            return currentClient.Endpoint;
```

Comment:
You can return `currentClient?.Endpoint`. This makes sure that if no matching client is found, this method would return null Uri instead of throwing null reference exception.